w_common/lib/src/common/disposable.dart

```diff
@@ -534,17 +534,20 @@ class Disposable implements _Disposable, DisposableManagerV7, LeakFlagger {
     });
     _internalDisposables.add(disposable);
 
-    completer.future.catchError((e) {
-      if (!_isDisposedOrDisposing) {
-        _logUnmanageMessage(completer);
-        _internalDisposables.remove(disposable);
-      }
-    }).then((_) {
-      if (!_isDisposedOrDisposing) {
-        _logUnmanageMessage(completer);
-        _internalDisposables.remove(disposable);
-      }
-    });
+    completer.future
+        // Log and remove disposable regardless of the outcome of the future.
+        .whenComplete(() {
+          if (!_isDisposedOrDisposing) {
+            _logUnmanageMessage(completer);
+            _internalDisposables.remove(disposable);
+          }
+        })
+        // Transform the stream from Future<T> to Future<Null> to allow catchError to exist.
+        // Without this, catchError would throw a TypeError at runtime.
+        .then((value) => null)
+        // Avoid uncaught errors in the isolate spawned by the whenComplete.
```

Comment:
I wasn't really familiar with the term isolate when we were talking yesterday, so I looked it up.  I don't think it is the correct term for this situation.  Each Future is just an event on the loop, all running in the main isolate.  https://medium.com/dartlang/dart-asynchronous-programming-isolates-and-event-loops-bffc3e296a6a 

If I'm not mistaken, then the terminology of this comment could update to be more correct.